project2-beat-that/script.js

```diff
@@ -1,4 +1,299 @@
+      output = `Player 1 number is ${playerOneNumber}. <br> 
+      Player 2 please press submit to generate your numbers `
+      return output;
+    }
+    else if (input == `SCORE`){
+      gameState = SCORE;
+    }
+    else if (input == `LOWEST`){
+      gameState = LOWEST_COMBINE_NUMBDER;
+    }
+    else{
+      return `Invalid response, key in "1" or "2"`;
+    }
+  }
+
+  // PLAYER 2 STARTS 
+  if (gameState == CHANGE_TO_PLAYER_2){
+    console.log(`Player 2 mode`);
+    gameState = CHOOSE_DICE_ORDER_PLAYER_2;
+    return playerTwoRollDice();
+  }
+
+// player 2 selects order 
+  if (gameState == CHOOSE_DICE_ORDER_PLAYER_2){
+    console.log(`New game state: ${gameState}`);
+    if (input == '1'){
+      gameState = COMPARE_PLAYERS;
+      playerTwoNumber = String(playerTwo[0]) + String(playerTwo[1]);
+      playerTwoCombine.push(playerTwoNumber);
+      output = `Player 1 number is ${playerOneNumber}. <br>
+      Player 2 number is ${playerTwoNumber}. <br>
+      Press submit to find out who wins!`
+      return output;
+    }
+    else if (input == '2'){
+      gameState = COMPARE_PLAYERS;
+      playerTwoNumber = String(playerTwo[1]) + String(playerTwo[0]);
+      playerTwoCombine.push(playerTwoNumber);
+      output = `Player 1 number is ${playerOneNumber}. <br>
+      Player 2 number is ${playerTwoNumber}. <br>
+      Press submit to find out who wins! `
+      return output;
+    }
+    else if (input == `SCORE`){
+      gameState = SCORE;
+    }
+   
+    else{
+      return `Invalid response, key in "1" or "2"`;
+    }
+  }
+
+  // COMPARE WINNER
+  if ( gameState == COMPARE_PLAYERS){
+    if (Number(playerTwoNumber) < Number(playerOneNumber)){
+      winner = `Player One`
+      output = `The winner is ${winner}. <br> Player 1 number is ${playerOneNumber}. <br>
+      Player 2 number is ${playerTwoNumber}. <br> `
+      if (input == `SCORE`){
+        gameState = SCORE;
+      }
+      else {
+        gameState = CONTINUE_GAME; 
+      }
+      
+      gameState = CONTINUE_GAME; 
+      return output;
+    }
+    else if (Number(playerTwoNumber) > Number(playerOneNumber)){
+      winner = `Player Two`
+      output = `The winner is ${winner}. <br> Player 1 number is ${playerOneNumber}. <br>
+      Player 2 number is ${playerTwoNumber}. <br> `
+      
+      gameState = CONTINUE_GAME; 
+      return output;
+    }
+    else if (Number(playerTwoNumber) == Number(playerOneNumber)){
+      winner = `Draw`
+      output = `The result is a ${winner}. <br> Player 1 number is ${playerOneNumber}. <br>
+      Player 2 number is ${playerTwoNumber}. <br> `
+      
+      gameState = CONTINUE_GAME; 
+      return output;
+    }
+   
+  }
+
+  // CONTINUE GAME
+  if (gameState == CONTINUE_GAME){
+  counter = 0; 
+  while (counter < 2){
+    playerOne.pop();
+    playerTwo.pop();
+    counter +=1; 
+  }
+  gameState = ROLL_DICE;
+    output = `Press submit to continue playing! <br> `; 
+    return output;  
+  }
```

Comment:
is this to reset the game? you can just do playerOne = [] and similarly for playerTwo to clear out the array rather than having the while loop

also consider making this a helper function